homeassistant/components/plex/config_flow.py

```diff
@@ -254,22 +258,55 @@ class PlexOptionsFlowHandler(config_entries.OptionsFlow):
     def __init__(self, config_entry):
         """Initialize Plex options flow."""
         self.options = copy.deepcopy(config_entry.options)
+        self.server_id = config_entry.data[CONF_SERVER_IDENTIFIER]
 
     async def async_step_init(self, user_input=None):
         """Manage the Plex options."""
         return await self.async_step_plex_mp_settings()
 
     async def async_step_plex_mp_settings(self, user_input=None):
         """Manage the Plex media_player options."""
+        plex_server = self.hass.data[DOMAIN][SERVERS][self.server_id]
+
         if user_input is not None:
             self.options[MP_DOMAIN][CONF_USE_EPISODE_ART] = user_input[
                 CONF_USE_EPISODE_ART
             ]
             self.options[MP_DOMAIN][CONF_SHOW_ALL_CONTROLS] = user_input[
                 CONF_SHOW_ALL_CONTROLS
             ]
+            self.options[MP_DOMAIN][CONF_IGNORE_NEW_SHARED_USERS] = user_input[
+                CONF_IGNORE_NEW_SHARED_USERS
+            ]
+
+            account_data = {}
+            for user in plex_server.accounts:
+                account_data.setdefault(
+                    user, {"enabled": bool(user in user_input[CONF_MONITORED_USERS])},
+                )
+            self.options[MP_DOMAIN][CONF_MONITORED_USERS] = account_data
+
             return self.async_create_entry(title="", data=self.options)
 
+        available_accounts = {name: name for name in plex_server.accounts}
+        available_accounts[plex_server.owner] += " [Owner]"
+
+        default_accounts = plex_server.accounts
+        known_accounts = set(self.options[MP_DOMAIN].get(CONF_MONITORED_USERS, []))
+        if known_accounts:
+            default_accounts = {
```

Comment:
These are not default_accounts but rather enabled_accounts?

Reply:
I actually had this named `enabled_accounts` before. But it's only used to create the list of defaults to the user so this felt a bit more appropriate.